# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/webkitgtk41/default-uri.patch
# Copyright (C) 2021 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.h.vanilla     2021-07-29 17:05:38.096130563 +0200
+++ webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.h     2021-07-29 17:05:57.046129807 +0200
@@ -37,7 +37,7 @@
 #define BROWSER_IS_WINDOW(obj)         (G_TYPE_CHECK_INSTANCE_TYPE((obj), BROWSER_TYPE_WINDOW))
 #define BROWSER_IS_WINDOW_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE((klass),  BROWSER_TYPE_WINDOW))
 #define BROWSER_WINDOW_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS((obj),  BROWSER_TYPE_WINDOW, BrowserWindowClass))
-#define BROWSER_DEFAULT_URL            "http://www.webkitgtk.org/"
+#define BROWSER_DEFAULT_URL            "http://t2sde.org/"
 #define BROWSER_ABOUT_SCHEME           "minibrowser-about"

 typedef struct _BrowserWindow        BrowserWindow;
--- webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.c.vanilla     2021-07-29 17:27:47.389077560 +0200
+++ webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.c     2021-07-29 17:29:07.242074376 +0200
@@ -94,13 +94,20 @@

 static void activateUriEntryCallback(BrowserWindow *window)
 {
-    browser_window_load_uri(window,
+    const char* uri =
 #if GTK_CHECK_VERSION(3, 98, 0)
-        gtk_editable_get_text(GTK_EDITABLE(window->uriEntry))
+        gtk_editable_get_text(GTK_EDITABLE(window->uriEntry));
 #else
-        gtk_entry_get_text(GTK_ENTRY(window->uriEntry))
+        gtk_entry_get_text(GTK_ENTRY(window->uriEntry));
 #endif
-    );
+    const char* proto = strstr(uri, "://");
+    if (!proto) {
+      char uri2[strlen("https://") + strlen(uri) + 1];
+      snprintf(uri2, sizeof(uri2), "https://%s", uri);
+      browser_window_load_uri(window, uri2);
+    } else {
+      browser_window_load_uri(window, uri);
+    }
 }

 static void reloadOrStopCallback(GSimpleAction *action, GVariant *parameter, gpointer userData)
